### DISCUSSION TITLE
Release notes for github.com/aws/aws-sdk-go-v2 go packages

### DISCUSSION THREAD

**Discussion by colinodell** [renovatebot/renovate#26597] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

[github.com/aws/aws-sdk-go-v2](https://github.com/aws/aws-sdk-go-v2) is a monorepo containing ~380 Go packages, each having its own individual `CHANGELOG.md` file:

| Package | `CHANGELOG.md` file |
| --- | --- |
| github.com/aws/aws-sdk-go-v2/config | https://github.com/aws/aws-sdk-go-v2/blob/main/config/CHANGELOG.md |
| github.com/aws/aws-sdk-go-v2/codegen | https://github.com/aws/aws-sdk-go-v2/blob/main/codegen/CHANGELOG.md |
| github.com/aws/aws-sdk-go-v2/credentials | https://github.com/aws/aws-sdk-go-v2/blob/main/credentials/CHANGELOG.md |
| github.com/aws/aws-sdk-go-v2/service/accessanalyzer | https://github.com/aws/aws-sdk-go-v2/blob/main/service/accessanalyzer/CHANGELOG.md |
| github.com/aws/aws-sdk-go-v2/service/account | https://github.com/aws/aws-sdk-go-v2/blob/main/service/account/CHANGELOG.md |
| github.com/aws/aws-sdk-go-v2/service/acm | https://github.com/aws/aws-sdk-go-v2/blob/main/service/acm/CHANGELOG.md |
| github.com/aws/aws-sdk-go-v2/service/acmpca | https://github.com/aws/aws-sdk-go-v2/blob/main/service/acmpca/CHANGELOG.md |
| github.com/aws/aws-sdk-go-v2/service/alexaforbusiness | https://github.com/aws/aws-sdk-go-v2/blob/main/service/alexaforbusiness/CHANGELOG.md |
| etc | ... |

Currently, **Renovate does not surface any of these changelogs** in the PRs it opens:

![image](https://github.com/renovatebot/renovate/assets/202034/f419ea14-1b6e-40dd-830d-e7ec25d57883)

I believe this is because:

1. The maintainers don't create individual releases with release notes for each package and version
2. The changelogs exist in separate individual subdirectories

**I therefore have to manually locate and read these changelogs whenever Renovate suggests updates.  It would be great if Renovate could help automate that for me.**

How would you recommend solving this?  Like... I'd be happy to open a PR adding all the changelogs to `metadata-manual.ts`, which would at least put changelogs links in the PRs:

![image](https://github.com/renovatebot/renovate/assets/202034/b02cfc94-b87e-48ca-95f2-f8bd0b7cbfd0)

But then y'all would have to keep updating that as AWS unveils new services.  Maybe it would be better to add some custom logic to dynamically generate those changelog URLs for this particular monorepo? If so, I'd be happy to contribute that if you don't mind offering some guidance on where/how you'd like that implemented :)

**Comment by colinodell** [comment]
Perhaps instead of [exporting the `manualChangelogUrls` map](https://github.com/renovatebot/renovate/blob/0b3b56adb555ae993a8f89ea721c5dfa36c4b259/lib/modules/datasource/metadata-manual.ts#L3C1-L3C33) we could export a function that attempts to return a hard-coded value from that map OR a dynamically-generated value for cases like this?

**Comment by rarkins** [comment]
If there is a naming convention then we can hopefully detect them automatically with an enhancement of our logic - we already have similar for JS monorepos. Can you create a reproduction with maybe 2 different modules needing updating?

**Reply by colinodell** [reply]
Yep, there is a naming convention they all follow:

```
github.com/aws/aws-sdk-go-v2/{some-package} => https://github.com/aws/aws-sdk-go-v2/blob/main/{some-package}/CHANGELOG.md
```

(where `{some-package}` may contain 0 or more `/`s)

And here's a reproduction: https://github.com/colinodell/renovate-repro-26597/pull/4

**Reply by rarkins** [reply]
Converted to issue: https://github.com/renovatebot/renovate/issues/26598
